yarn-project/simulator/src/public/executor.ts

```diff
@@ -169,4 +172,82 @@ export class PublicExecutor {
     const newWorldState = context.persistableState.flush();
     return temporaryConvertAvmResults(execution, newWorldState, result);
   }
+
+  /**
+   * These functions are currently housed in the temporary executor as it relies on access to
+   * oracles like the contractsDB and this is the least intrusive way to achieve this.
+   * When we remove this executor and have an interface that is compatible with the kernel circuits,
+   * this will be moved to sequencer-client/prover.
+   */
+
+  /**
+   * Generates a proof for an associated avm execution. This is currently only used for testing purposes,
+   * as proof generation is not fully complete in the AVM yet.
+   * @param execution - The execution to run.
+   * @returns An AVM proof and the verification key.
+   */
+  public async getAvmProof(avmExecution: PublicExecution): Promise<Buffer[]> {
+    // The paths for the barretenberg binary and the write path are hardcoded for now.
+    // We additionally need the path to a valid crs for proof generation.
+    const bbPath = '../../barretenberg/cpp';
```

Comment:
Something interesting here: I think right now the TS tree (and noir?) really only depends on bb wasm. This introduces a dependency to the bb binary. It might be good to be very explicit about this in the PR description.

For bootstrap: this will get built.

For the CI or other workflows: not sure! please double check.

Reply:
@fcarreiro makes a good point here :thinking: Maybe @Maddiaa0 can provide some guidance regarding an intermediate path forward here. If it makes sense for the simulator to have a dependency on the bberg main binary, then he should be able to provide guidance for getting that set up in CI.

I think it's okay to get this PR merged and clean this up in a follow-up PR. But I don't think we should leave this as-is and move on. Only merge with the intention of revisiting shortly in a follow-up PR!